### ISSUE TITLE
Meta not being added to vector on v1.1.5 - works on v1.1.3

### ISSUE THREAD

**Issue by boschi** [juxt/aero#86]
Meta not being added to vector on v1.1.5 - works on v1.1.3

Example:
```clojure
{:post [^:interceptors
    [#interceptor :my.company.interceptors.auth/parse-jwt
     #interceptor :my.company.interceptors.auth/verify-signature
     #interceptor [:my.company.interceptors.auth/scope? {:scope "admin/csv-import"}]
     #interceptor [:my.company.interceptors.params/coerce-params {:body-params {:filename string?}}]]
    #handler :my.company.handlers.import/trigger-csv-import]}
```

Then when we query for the `meta` on the vector that is the value of :post, on aero v1.1.3 it returns `:interceptors`, as expected, however on aero v1.1.5 it returns `nil`

**Comment by SevereOverfl0w** [comment]
I would guess reconstruction isn't functioning correctly.

**Comment by SevereOverfl0w** [comment]
Vectors are easy to fix, so I have that sussed. However I've noticed lists are also affected by this.  This is caused by postwalk dropping the metadata.

**Comment by SevereOverfl0w** [comment]
Fixed in 1.1.6. Thanks for opening the issue!
